src/field/pokemon.ts

```diff
@@ -312,128 +344,79 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
-      this.evoCounter = dataSource.evoCounter ?? 0;
-      this.usedTMs = dataSource.usedTMs ?? [];
-      this.resetCustomPokemonData(dataSource.customPokemonData);
-      this.teraType = dataSource.teraType;
-      this.isTerastallized = dataSource.isTerastallized;
-      this.stellarTypesBoosted = dataSource.stellarTypesBoosted ?? [];
-    } else {
-      this.id = globalScene.getNextPokemonID();
-      this.personalityValue = this.randSeedInt(Math.pow(2, 32) - 1);
-      this.ivs = ivs || this.generateIvs();
-
-      if (this.gender === undefined) {
-        this.generateGender();
-      }
+    // This may increment `nextPokemonID` twice, but shouldn't have any adverse effects...
+    this.id = options.id ?? globalScene.getNextPokemonID();
+    globalScene.updateNextPokemonID(this.id);
 
-      if (this.formIndex === undefined) {
-        this.formIndex = globalScene.getSpeciesFormIndex(species, this.gender, this.nature, this.isPlayer());
-      }
+    this.nickname = options.nickname ?? null;
+    this.personalityValue = options.personalityValue ?? this.randSeedInt(Math.pow(2, 32) - 1);
 
-      if (this.shiny === undefined) {
-        this.trySetShiny();
-      }
+    this.gender = options.gender ?? this.generateGender();
 
-      if (this.variant === undefined) {
-        this.variant = this.shiny ? this.generateShinyVariant() : 0;
-      }
+    this.shiny = options.shiny ?? this.trySetShiny();
+    this.variant = options.variant ?? this.generateShinyVariant();
 
-      this.resetCustomPokemonData();
+    this.formIndex =
+      options.formIndex ?? globalScene.getSpeciesFormIndex(species, this.gender, this.nature, this.isPlayer());
 
-      if (nature !== undefined) {
-        this.setNature(nature);
-      } else {
-        this.generateNature();
-      }
+    this.resetCustomPokemonData(options.customPokemonData);
 
-      this.friendship = species.baseFriendship;
-      this.metLevel = level;
-      this.metBiome = globalScene.currentBattle ? globalScene.arena.biomeId : -1;
-      this.metSpecies = species.speciesId;
-      this.metWave = globalScene.currentBattle ? globalScene.currentBattle.waveIndex : -1;
-      this.pokerus = false;
+    this.ivs = options.ivs ?? this.generateIvs();
+    this.nature = options.nature ?? randSeedItem(Object.values(Nature));
+    if (options.stats != null) {
+      this.stats = options.stats;
+    } else {
+      this.calculateStats();
     }
+    this.hp = options.hp ?? this.getMaxHp();
 
-    this.generateName();
+    this.abilityIndex = options.abilityIndex ?? this.generateRandomAbility();
+    this.passive = options.passive ?? false;
 
-    if (!dataSource) {
-      this.calculateStats();
+    if (isPokemon(options)) {
+      this.moveset = options["moveset"];
+      this.status = options["status"];
+    } else {
+      // TODO: What should be the default value here?
+      this.moveset = options.moveset ?? [];
+      this.status = options.status ?? null;
     }
```

Comment:
```suggestion
    this.moveset = options["moveset"] ?? [];
    this.status = options["status"] ?? null;
```
The `if/else` isn't necessary since `moveset` and `status` ended up being part of `PokemonOptions`.

Also, the moveset should be constructed in either `PlayerPokemon` or `EnemyPokemon` if not defined here, I believe, so defaulting to `[]` is fine (isn't that what it already does?).